src/google/protobuf/io/coded_stream.h

```diff
@@ -1349,6 +1349,8 @@ inline CodedInputStream::CodedInputStream(const uint8* buffer, int size)
     extension_factory_(NULL) {
   // Note that setting current_limit_ == size is important to prevent some
   // code paths from trying to access input_ and segfaulting.
+  if(total_bytes_limit_ < size)
```

Comment:
How about changing the line in the initializer list instead? I.e., change:
```cpp
total_bytes_limit(kDefaultTotalBytesLimit),
```
to
```cpp
total_bytes_limit(max(size, kDefaultTotalBytesLimit)),
```